### DISCUSSION TITLE
How SSO integration can be done on Dependency Track Application implemented on Nomad

### DISCUSSION THREAD

**Discussion by kiranmahale24** [DependencyTrack/dependency-track#2399] [Q&A]
Hello team,

We have deployed the Dependency Track application on Nomad.
I am able to login as admin.

Now I want to integrate SSO login in Dependency track.
Can you please help me and suggest how it can be integrated.

**Comment by kiranmahale24** [comment]
Hello @stevespringett ,
Can you please help me with this topic?

**Comment by nscuro** [comment]
Dependency-Track supports both LDAP and OIDC for SSO. Refer to the documentation for details:

* https://docs.dependencytrack.org/getting-started/ldap-configuration/
* https://docs.dependencytrack.org/getting-started/openidconnect-configuration/

If that doesn't fit your needs, please elaborate on what you need for integration with Nomad. AFAIK, Hashicorp Nomad is a workload orchestration tool and does not itself provide any SSO capabilities. But I may be wrong.

**Reply by kiranmahale24** [reply]
I have gone through the documentation for both LDAP and Open ID.
 
My Scenario:- Dependency track deployed on Nomad. Able to login with admin.
New Activity: Enable SSO Login for Dependency track

Option 1: Open ID
I have gone through the Open ID configurations. Do I need to have third party like Keycloak to add configuration settings?

Option 2: LDAP
For LDAP, the below configurations are to be added where:
alpine.ldap.enabled=true
alpine.ldap.server.url=ldap://ldap.example.com:3268
alpine.ldap.basedn=dc=example,dc=com
alpine.ldap.security.auth=simple
alpine.ldap.auth.username.format=%s@example.com
alpine.ldap.bind.username=cn=ServiceAccount,ou=Users,dc=example,dc=com
alpine.ldap.bind.password=mypassword
alpine.ldap.attribute.name=userPrincipalName
alpine.ldap.attribute.mail=mail
alpine.ldap.groups.filter=(&(objectClass=group)(objectCategory=Group))
alpine.ldap.user.groups.filter=(member:1.2.840.113556.1.4.1941:={USER_DN})
alpine.ldap.groups.search.filter=(&(objectClass=group)(objectCategory=Group)(cn=*{SEARCH_TERM}*))
alpine.ldap.users.search.filter=(&(objectClass=user)(objectCategory=Person)(cn=*{SEARCH_TERM}*))

**Reply by nscuro** [reply]
Both OIDC and LDAP require an external service, be it self-hosted or SaaS.

Regarding where the properties are supposed to be put, if you're running the Docker containers of DT, you can use environment variables. See https://docs.dependencytrack.org/getting-started/deploy-docker/#docker-compose-automated--orchestration

**Reply by kiranmahale24** [reply]
Hello @nscuro ,
After setting the ldap config in env , how to make a call to these ldap definitions for the application to understand these config .

**Reply by kiranmahale24** [reply]
Hello @nscuro @stevespringett ,
We have added as an environment the LDAP properties as below:
- ALPINE_LDAP_ENABLED=true
     - ALPINE_LDAP_SERVER_URL=
     - ALPINE_LDAP_BASEDN=
     - ALPINE_LDAP_SECURITY_AUTH=simple
     - ALPINE_LDAP_BIND_USERNAME=
     - ALPINE_LDAP_BIND_PASSWORD=
     - ALPINE_LDAP_AUTH_USERNAME_FORMAT=
     - ALPINE_LDAP_ATTRIBUTE_NAME=displayname
     - ALPINE_LDAP_ATTRIBUTE_MAIL=mail
     - ALPINE_LDAP_GROUPS_FILTER=(&(objectClass=group)(objectCategory=Group))
     - ALPINE_LDAP_USER_GROUPS_FILTER=(member:1.2.840.113556.1.4.1941:={USER_DN})
     - ALPINE_LDAP_GROUPS_SEARCH_FILTER=(&(objectclass=group)(cn={0}))
     - ALPINE_LDAP_USERS_SEARCH_FILTER=(&(objectClass=user)(objectCategory=Person)(cn={SEARCH_TERM}))
     - ALPINE_LDAP_USER_PROVISIONING=false
     - ALPINE_LDAP_TEAM_SYNCHRONIZATION=false

After adding user in LDAP User of Dependency Track; it is still showing as Syncing.
![LDAP-sync](https://user-images.githubusercontent.com/123181711/217892096-23d4aa52-4288-4353-9a13-e237b9b52ac2.png)

Can you please suggest why am getting Syncing status.
